lifecycle/lifecycle/endpoints/auth.py

```diff
@@ -104,3 +135,30 @@ def _generate_tokens_for_all_escs(request: Request):
         check_auth(request, scope=AuthScope.CALL_ADMIN_API)
         regenerate_all_esc_tokens()
         
+
+def _normalize_endpoint_path(endpoint: str) -> str:
+    if endpoint.endswith('/'):
+        endpoint = endpoint[:-1]
+    if not endpoint.startswith('/'):
+        endpoint = '/' + endpoint
+    return endpoint
+
+
+def _authorize_job_caller(job_model: models.Job, endpoint: str, request: Request):
```

Comment:
Nitpick: looking at just the function definition in the MR, I had problem understanding how it accepts and rejects the requests, I had to open IDE and lookup the internal functions. So I think adding short comment about the thrown exception in case of rejection etc could be useful.